File: app/src/main/java/com/exam/main/contract/MainContract.java
package com.exam.main.contract;

/**
 * @author LowAndroider
 * @date 15:06
 */

public interface MainContract {
    interface Model {
        /**
         * 设置当前页面的标识
         */
        void setCurrentIndex(int index);

        /**
         * @return 当前页面的标识
         */
        int getCurrentIndex();
    }

    interface View {

        /**
         * 切换页面
         * @param index
         */
        void switchCard(int index);
    }

    interface Presenter {
        /**
         * 设置当前页面的标识
         */
        void setCurrentIndex(int index,View view);
    }
}

File: app/src/main/java/com/exam/main/contract/MainContract.kt
package com.exam.main.contract

/**
 * @author LowAndroider
 * @date 15:06
 */

interface MainContract {
    interface Model {

        /**
         * @return 当前页面的标识
         */
        /**
         * 设置当前页面的标识
         */
        var currentIndex: Int
    }

    interface View {

        /**
         * 切换页面
         * @param index
         */
        fun switchCard(index: Int)
    }

    interface Presenter {
        /**
         * 设置当前页面的标识
         */
        fun setCurrentIndex(index: Int, view: View)
    }
}
